src/lib.rs

```diff
@@ -68,17 +68,40 @@ impl PixelSelector {
     }
 }
 
-trait ParticularValues: From<f32> + Zero + One {
-    fn ceil(&self) -> Self;
-}
-impl ParticularValues for f32 {
-    fn ceil(&self) -> Self {
-        1e10f32
+trait ParticularValues: From<f32> + Zero + One {}
+impl ParticularValues for f32 {}
+impl ParticularValues for f64 {}
+
+fn time_to_next_pixel<
+    T: PartialOrd + Neg<Output = T> + Div<Output = T> + Sub<Output = T> + ParticularValues,
+>(
+    velocity: T,
+    current_frac: T,
+) -> T {
+    if velocity > 0.0.into() {
+        let one: T = 1.0.into();
+        (one - current_frac) / velocity
+    } else if velocity < 0.0.into() {
+        -(current_frac / velocity)
+    } else {
+        std::f32::INFINITY.into()
```

Comment:
note that I just ran `cargo clippy` and discovered this name was deprecated. Fixed in e69c4fd210db04eeb9aca4f93e69a244d009a478